src/client/context.ts

```diff
@@ -35,22 +36,60 @@ export interface Context {
     id?: {[key: string]: string|undefined;};
 
     /**
+     * @hidden
      * A context object is open for extension with any custom properties/metadata.
      */
     [key: string]: unknown;
 }
 
-// Built-in contexts
+/**
+ * Built-in context to define a contact.
+ */
 export interface ContactContext extends Context {
+    /**
+     * The context type is always 'fdc3.contact'.
+     */
     type: 'fdc3.contact';
+    /**
+     * Name of the contact.
+     */
     name: string;
+    /**
+     * The contact data. Currently has optional fields for email address, Twitter handle, and phone number.
+     */
     id: {[key: string]: string}&{email?: string; twitter?: string; phone?: string};
```

Comment:
Again, not 100% about the docs, but this weird type intersection syntax generates docs with the definition `id: object & object` which look a bit weird. Is there a better way of handling this syntactically that would make the docs look a bit better?

Reply:
I have no idea about this one, sorry

Reply:
You get 

`Property 'email' of type 'string | undefined' is not assignable to string index type 'string'.`

Reply:
Been playing around with this some more. Not possible to combine optional properties with `[key: string]: string`, and cannot list individual `id` fields within the generated TypeScript docs without creating a separate type, which seems a bit bloat-y.

IMO, rely on listing the individual fields within the `id` comment (as currently), and on IDE intellisense.